cupy/core/_scalar.pyx

```diff
@@ -85,6 +85,9 @@ cdef _setup_type_dict():
         _typenames[t] = _typenames_base[d]
         k = ord(d.kind)
         _dtype_kind_size_dict[t] = (k, d.itemsize)
+    # CUDA types
+    for t in ('cudaTextureObject_t',):
+        _typenames[t] = t
```

Comment:
Leaving a note before I forgot what's going on here: the key could perhaps be renamed to a different one from the value (`cudaTextureObject_t`) used in actual code -- say, for example, `CUPY_TexObj` -- if we wanna emphasize it's an internal type used by CuPy.